test/convergence.jl

```diff
@@ -39,10 +39,23 @@ for (prob, sol, tscale) in [
 
 end
 
-@test convergence_order(split_linear_prob_wfact_split, linear_sol, ARS111(linsolve=linsolve_direct), dts) ≈ 1 atol=0.05
-@test convergence_order(split_linear_prob_wfact_split, linear_sol, ARS121(linsolve=linsolve_direct), dts) ≈ 1 atol=0.05
-@test convergence_order(split_linear_prob_wfact_split, linear_sol, ARS232(linsolve=linsolve_direct), dts) ≈ 2 atol=0.05
-@test convergence_order(split_linear_prob_wfact_split, linear_sol, ARS343(linsolve=linsolve_direct), dts) ≈ 3 atol=0.05
+algs1 = (ARS111, ARS121)
+algs2 = (ARS122, ARS232, ARS222)
+algs3 = (ARS233, ARS343, ARS443, IMKG342a, IMKG343a, IMKG353a)
+for (algorithm_names, order) in ((algs1, 1), (algs2, 2), (algs3, 3))
+    for algorithm_name in algorithm_names
+        for (problem, solution) in (
+            (split_linear_prob_wfact_split1, linear_sol),
+            (split_linear_prob_wfact_split2, linear_sol),
+        )
+            algorithm = algorithm_name(
+                NewtonsMethod(; linsolve = linsolve_direct, max_iters = 2),
+            ) # setting max_iters = 1 results in incorrect convergence orders
+            @test convergence_order(problem, solution, algorithm, dts) ≈ order atol=0.01
+        end
+    end
+end
```

Comment:
Depending on how long this takes, we could add CL args and run these in parallel.

Reply:
This is a linear initial-value problem with a single variable, so it actually runs in only a second or two after compilation.